src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,149 @@
+package main.java.Duke;
+
+import main.java.Duke.Commands.*;
+import main.java.Duke.DukeException.DukeArrayException;
+import main.java.Duke.DukeException.DukeException;
+import main.java.Duke.Task.Deadline;
+import main.java.Duke.Task.Event;
+import main.java.Duke.Task.TaskList;
+import main.java.Duke.Task.toDo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    Duke duke;
+
+    public Parser(Duke duke) {
+        this.duke = duke;
+    }
+
+     /**
+     * Parses the user input and returns a Command to be executed.
+     *
+     * @param userInput User input from the user.
+     * @param tasklist Current task list.
+     * @return Command to be executed.
+     * @throws DateTimeParseException if user input for date is not in the correct format.
+     * @throws DukeException if user input is invalid.
+     */
+     public Command parse(String userInput, TaskList tasklist) {
```

Comment:
One neater alternative would be to make smaller methods such as handleDeadline() or handleEvent() for example instead of one huge method which handles every case bulked together. This could make the code appear untidy and bulky.